src/node_task_runner.h

```diff
@@ -0,0 +1,53 @@
+#ifndef SRC_NODE_TASK_RUNNER_H_
+#define SRC_NODE_TASK_RUNNER_H_
+
+#if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
+
+#include "node_internals.h"
+#include "simdjson.h"
+#include "spawn_sync.h"
+#include "uv.h"
+
+#include <optional>
+#include <string_view>
+
+namespace node {
+namespace task_runner {
+
+class ProcessRunner {
```

Comment:
Nit: some code comment documentation of what this is would be helpful. I know we're not too good or consistent on documenting our header files, etc but someone coming into the codebase fresh may have a difficult time understanding what this is used for.

Reply:
Well this is a first. I'll definitely add more comments! I'm more than happy to! :-)

Reply:
heh, it's something I've been wanting for a while... we need lots more comments throughout our code

Reply:
I thought comments were discouraged? Almost all the C++ code is comment-less and I assumed that's what was expected.

Reply:
There was once a bit of mindshare in Node.js for the "code should be self-documenting" idea, but fortunately I think that has mostly gone away. There's too much complexity in a lot of the Node.js core code for that to be reasonable and make any amount of sense to newbies. 😅